src/backend/distributed/executor/local_executor.c

```diff
@@ -468,48 +468,45 @@ ExtractLocalAndRemoteTasks(bool readOnly, List *taskList, List **localTaskList,
 		/* either the local or the remote should be non-nil */
 		Assert(!(localTaskPlacementList == NIL && remoteTaskPlacementList == NIL));
 
-		if (list_length(task->taskPlacementList) == 1)
+		if (list_length(localTaskPlacementList) > 0)
 		{
-			/*
-			 * At this point, the task has a single placement (e.g,. anchor shard
-			 * is distributed table's shard). So, it is either added to local or
-			 * remote taskList.
-			 */
-			if (localTaskPlacementList == NIL)
+			if (list_length(task->taskPlacementList) == 1)
```

Comment:
I think there's nothing particularly interesting about tasks with a single placement.

How about we do something like this?

```c
        if (localTaskPlacementList == NIL)
        {
            *remoteTaskList = lappend(*remoteTaskList, task);
        }
        else if (remoteTaskPlacementList == NIL)
        {
            *localTaskList = lappend(*localTaskList, task);
        }
        else
        {
            /*
             * At this point, we're dealing with a task that has placements on both 
             * local and remote nodes.
             */
            task->partiallyLocalOrRemote = true;

```

Reply:
This simplifies it further, much easier to understand. Thanks for the idea.